CI fails for 3.0.x and 2.2.x due to recent pytest version

### Bug report

CI fails on branches 3.0.x and 2.2.x fails with

~~~
E               _pytest.warning_types.RemovedInPytest4Warning: MarkInfo objects are deprecated as they contain merged marks which are hard to deal with correctly.
E               Please use node.get_closest_marker(name) or node.iter_markers(name).
E               Docs: https://docs.pytest.org/en/latest/mark.html#updating-code
~~~

See e.g. #12811, #12812.

This is due to a recent pytest version.

Solution: Either pin pytest<3.8 (I think that's the correct number) or backport #12154 (and maybe  #12297).

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -66,7 +66,8 @@ install:
   - activate test-environment
   - echo %PYTHON_VERSION% %TARGET_ARCH%
   # pytest-cov>=2.3.1 due to https://github.com/pytest-dev/pytest-cov/issues/124
-  - pip install -q "pytest>=3.4" "pytest-cov>=2.3.1" pytest-rerunfailures pytest-timeout pytest-xdist
+  - pip install -q "pytest>=3.4,<3.8" "pytest-cov>=2.3.1" pytest-rerunfailures
+    pytest-timeout pytest-xdist
 
   # Apply patch to `subprocess` on Python versions > 2 and < 3.6.3
   # https://github.com/matplotlib/matplotlib/issues/9176

--- a/requirements/testing/travis_all.txt
+++ b/requirements/testing/travis_all.txt
@@ -8,7 +8,8 @@ pillow
 pyparsing
 # pytest-timeout master depends on pytest>=3.6. Testing with pytest 3.4 is
 # still supported; this is tested by the first travis python 3.5 build
-pytest>=3.6
+# pytest>=3.8 throws RemovedInPytest4Warning (github #12825)
+pytest>=3.6,<3.8
 pytest-cov
 pytest-faulthandler
 pytest-rerunfailures

